sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/HashProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SonarSource :: .NET :: Shared library
+ * Copyright (C) 2014-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.dotnet.shared.plugins;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class HashProvider {
+  public byte[] computeHash(Path filePath) throws NoSuchAlgorithmException, IOException {
```

Comment:
Open question: Isn't this one going to make our life easier? 
```suggestion
  public byte[] computeHash(Path filePath) throws Throwable {
```

Reply:
With this the contract is clear. We say exactly what kind of exceptions we throw. 

When handling the exception can use the base type `Throwable` or `Exception` or even be explicit, depending on the needs.